src/main/java/org/folio/search/service/converter/FacetQueryBuilder.java

```diff
@@ -46,16 +57,18 @@
     return request.getFacet().stream()
       .map(this::getFacetFieldAndLimitAsPair)
       .map(facet -> getFacetAggregation(request, query, facet))
+      .flatMap(Collection::stream)
       .collect(toList());
   }
 
-  private AggregationBuilder getFacetAggregation(CqlFacetServiceRequest request,
+  private List<AggregationBuilder> getFacetAggregation(CqlFacetServiceRequest request,
     QueryBuilder query, Pair<String, Integer> facetAndLimit) {
-    var field = facetAndLimit.getLeft();
+    var field = facetAndLimit.getFirst();
     validateFacetField(field, request.getResource());
-    return getFilterQueryForFacet(field, query)
-      .map(filter -> filterAggregation(filter, facetAndLimit))
-      .orElse(termsAggregation(field, facetAndLimit));
+    var filterAndFacetTerms = getFilterQueryAndFacetTerms(field, query);
+    return filterAndFacetTerms.getFirst() != null
```

Comment:
What does it mean when Facet name is null? Should we add a comment?

Reply:
It is not facet name, first value - is the filter query for facet, it can be null if user don't specify filer conditions